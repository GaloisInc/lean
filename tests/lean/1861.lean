def int' := int

#eval id_rhs int' (-1 : ℤ)
-- #2147483647

instance : has_repr int' :=
by unfold int'; apply_instance

#eval id_rhs int' (-1 : ℤ)
-- -1

@[reducible] def int'' := int

/- Don't need to define has_repr instance for int'' because it is reducible -/
#eval id_rhs int'' (-1 : ℤ)
-- -1

inductive foo
| mk₁ : bool → foo
| mk₂ : bool → foo

#eval foo.mk₁ tt
