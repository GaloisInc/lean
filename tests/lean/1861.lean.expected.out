1861.lean:3:0: warning: result type does not have an instance of type class 'has_repr', dumping internal representation
#2147483647
-1
-1
1861.lean:22:0: warning: result type does not have an instance of type class 'has_repr', dumping internal representation
(#0 #1)
